atlasdb-client/src/test/java/com/palantir/atlasdb/keyvalue/impl/TracingKeyValueServiceTest.java

```diff
@@ -0,0 +1,442 @@
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetRequest;
+import com.palantir.atlasdb.keyvalue.api.ColumnRangeSelection;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowColumnRangeIterator;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.tracing.TestSpanObserver;
+import com.palantir.common.base.ClosableIterator;
+import com.palantir.common.base.ClosableIterators;
+import com.palantir.remoting1.tracing.SpanType;
+import com.palantir.remoting1.tracing.Tracer;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TracingKeyValueServiceTest {
+
+    private KeyValueService delegate;
+    private KeyValueService kvs;
+
+    private final TestSpanObserver observer = new TestSpanObserver();
+    private final Namespace nameSpace = Namespace.create("test");
+    private final byte[] rowName = "row".getBytes(Charsets.UTF_8);
+    private final byte[] colName = "col".getBytes(Charsets.UTF_8);
+    private final TableReference tableRef = TableReference.create(nameSpace, "testTable");
+    private final Cell cell = Cell.create(rowName, colName);
+    private final RangeRequest rangeRequest = RangeRequest.all();
+    private final ImmutableList<RangeRequest> rangeRequests = ImmutableList.of(rangeRequest);
+    private final long timestamp = 1L;
+    private byte[] contents = "value".getBytes(Charsets.UTF_8);
+    private final Value value = Value.create(contents, timestamp);
+    private byte[] metadata = "metadata".getBytes(Charsets.UTF_8);
+
+    @Before
+    public void before() throws Exception {
+        Tracer.initTrace(Optional.of(true), getClass().getSimpleName() + "." + Math.random());
+        Tracer.subscribe(getClass().getName(), observer);
+        delegate = mock(KeyValueService.class);
+        kvs = TracingKeyValueService.create(delegate);
+        assertThat(observer.spans(), hasSize(0));
+    }
+
+    @After
+    public void after() throws Exception {
+        Tracer.unsubscribe(getClass().getName());
+        kvs.close();
+        verify(delegate, atLeast(1)).close();
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void createNullThrows() throws Exception {
+        TracingKeyValueService.create(null);
+    }
+
+    @Test
+    public void addGarbageCollectionSentinelValues() throws Exception {
+        ImmutableSet<Cell> cells = ImmutableSet.of(cell);
+        kvs.addGarbageCollectionSentinelValues(tableRef, cells);
+
+        checkSpan("atlasdb-kvs.addGarbageCollectionSentinelValues(test.testTable, 1 cells)");
```

Comment:
nit: `test.testTable` is repeated in several places, and is constructed from two magic strings that are also defined above. It could be declared as a constant.